### PR TITLE
openapi: Add test to ensure plus is maintained in Intl Phone Numbers

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
Add test to ensure plus is maintained in Intl Phone Numbers

## Related Issues
zaproxy/zaproxy#6644 

## Checklist
- [NA] Update help
- [NA] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

**Comment by thc202** [comment]
Does it fail with version 17?

**Comment by kingthorin** [comment]
I didn't check. I didn't think it was worth it. I guess there's a chance I'm not testing the right condition but based on the absence of feedback I waned something future thinking 😉

**Comment by thc202** [comment]
Just tried to import the definition of this PR with version 17 and it generates:
```json
{"name":"John Doe","tag":"John Doe","phoneNumber":"+15555555555"}
```

**Comment by kingthorin** [comment]
🤷‍♂️ I'm find to just close the PR and the issue. Just lemme know.

**Comment by thc202** [comment]
I'm good with the PR it is more how to close the issue, IMO we should close with insufficient evidence as we are not able to reproduce it.

**Comment by kingthorin** [comment]
Yup that makes sense. Thanks for talking it out, I didn't understand how you were looking at it.

File: addOns/openapi/src/test/resources/org/zaproxy/zap/extension/openapi/v3/petstore_reduced_intl_phone.yaml [hunk 1]
```diff
@@ -0,0 +1,160 @@
+openapi: "3.0.0"
```
Comment (thc202) [inline addOns/openapi/src/test/resources/org/zaproxy/zap/extension/openapi/v3/petstore_reduced_intl_phone.yaml:1]: I'd suggest trim to the absolutely minimum required for the test.

Comment (kingthorin) [reply addOns/openapi/src/test/resources/org/zaproxy/zap/extension/openapi/v3/petstore_reduced_intl_phone.yaml:1]: I'll be honest, I don't actually "know". I tried to trim it down to what seemed reasonable but I don't know the add-on or OpenAPI spec well enough.

Comment (thc202) [reply addOns/openapi/src/test/resources/org/zaproxy/zap/extension/openapi/v3/petstore_reduced_intl_phone.yaml:1]: Ok if I push some changes and you then fix up?

Comment (kingthorin) [reply addOns/openapi/src/test/resources/org/zaproxy/zap/extension/openapi/v3/petstore_reduced_intl_phone.yaml:1]: 100% 

Thank you!

Comment (thc202) [reply addOns/openapi/src/test/resources/org/zaproxy/zap/extension/openapi/v3/petstore_reduced_intl_phone.yaml:1]: Done.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by kingthorin** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**